tests/unit/helpers/DummyDatabase.php

```diff
@@ -16,24 +17,27 @@ public function find_registration_by_issuer($iss, $clientId = null)
             $registrations = json_decode($registrationDBFile, true);
             foreach ($registrations as $registrationDetails) {
                 if ($registrationDetails['issuer'] === $iss) {
-                    if (empty($clientId) || $registrationDetails['client_id'] === $clientId) {
+                    if (empty($clientId)
+                        || $registrationDetails['client_id'] === $clientId
+                        || $registrationDetails['aud'] === $clientId
```

Comment:
This assumed `client_id` in the test data which is opposite to the functional code (and resulted in pickling my brain, but I think all is now well?)